### ISSUE TITLE
Virheet ja näytteet

### ISSUE THREAD

**Issue by tpalonen** [NatLibFi/Finto-data#804]
TSK:n @kseppala :lta tuli palautetta. 

1) Virheet-käsitteen yläkäsite yleiset abstraktit objektit on poistunut YSOsta.
2) Näytteet-käsite: olisiko parempi käsite "käyttöön liittyvä rooli"? Nykyinen yläkäsite on elottomat objektit. Vrt. aineistot-käsite.

**Comment by tpalonen** [comment]
PÄÄTÖS 21.11.2016:

1) virheet YK abstraktit objektit

Roolit vs objektit -kysymys vaatisi perusteellista keskustelua. Nykyisin elottomissa objekteissa paljon sellaista, jonka voisi tulkita rooleiksi (esim. hylyt, romu). Rajanveto tässä vaikeaa. Esim. kalusteet tarkoittaa esineitä, joita käytetään kalusteina. Onko kalustekin siis rooli?

**Comment by kseppala** [comment]
Joo, vaatii tosiaan perusteellista tarkistamista. Ei ole kovin vaikeaa, kunhan miettii sitä, mikä jokin on ensisijaisesti (pääsääntöisesti). Kalusteet on tehty ensisijaisesti kalusteiksi, ei siis ole kyse roolista. Autoa ei ole tehty ensisijaisesti romuksi, mutta siitä voi tulla romua - on siis kyse roolista.

**Comment by tpalonen** [comment]
Virheet-käsitteen osalta tehty em. korjaus.

**Comment by kseppala** [comment]
Siis mikä oli virheen vanha yläkäsite ja mikä on uusi?

**Comment by tpalonen** [comment]
Virheet-käsitteen nykyinen yläkäsite on "abstraktit objektit" kuten on 21.11.2016 YSO-kokouksessa päätetty. Ennen tätä yläkäsite on ollut "yleiset abstraktit objektit", joka on ollut deprekoitu käsite. Ja vielä tätä aiemmin yläkäsitteenä oli "sisäiset ominaisuudet". Yläkäsite on vaihdettu nykyiseksi siis jo 2016.

**Comment by kseppala** [comment]
Ilman abstraktin objektin ja yleisen abstraktin objektin määritelmiä on vaikea varmasti sanoa, onko muutos ollut merkittävä, mutta isommasta muutoksesta on ollut kyse silloin, kun on käsite on siirretty ominaisuuksien hierarkiasta objektiksi.

**Comment by tpalonen** [comment]
Nyt mennään muistitiedon varassa, mutta "yleiset abstraktit objektit" oli muistaakseni lyhytikäiseksi jäänyt hierarkisoiva käsite, jonka käytännössä "abstraktit objektit" korvasi, koska "yleiset"-määrettä pidettiin tarpeettomana lisänä. Virheiden siirtyminen objekti-hierarkiaan taas liittyy ominaisuudet-hierarkian läpikäyntiin, jossa ominaisuudet määriteltiin uudella tavalla. Vrt. ominaisuuksien nykyinen määritelmä: _"Ominaisuudet luonnehtivat olioita ja tapahtumia. Ominaisuudet eivät ole olemassa itsenäisinä, vaan ne ”realisoituvat” muissa entiteeteissä. Esimerkkejä ominaisuuksista ovat muoto, paino, pituus, väri. Ominaisuuden instanssi (esim. tietty punaisuuden sävy) voi olla läsnä useassa objektissa samanaikaisesti."_ Pidimme uutta määritelmää vasten virheitä enemmän objektimaisena kuin ominaisuusmaisena käsitteenä.

**Comment by kseppala** [comment]
Tuon perusteella yläkäsitteen vaihtuminen yleisestä abstraktista objektista abstraktiksi objektiksi ei siis ole merkittävä. Koska ominaisuudet eivät ole em. määritelmän mukaan olemassa itsenäisinä ja abstraktit objektit taas ovat olemassa itsenäisinä, ko. muutos on ollut merkittävä. Omasta mielestäni virhe sopii em. määritelmän mukaan ominaisuuden alakäsitteeksi: virhe luonnehtii olioita ja tapahtumia, eikä ole olemassa itsenäisenä. Monet erilaiset toiminnat ja toiminnan lopputulokset voidaan tietyissä tilanteissa tulkita virheiksi, mutta ei tule mieleen esimerkkiä asiasta, joka olisi vain virhe. Toisaalta sama asia voidaan yhdessä tilanteessa katsoa virheeksi ja toisessa ei eli olosuhteet vaikuttavat siihen, mikä katsotaan virheeksi. Esimerkiksi ommel: jos se on tehty oikeaan paikkaan, kyse ei ole virheestä, mutta väärään kohtaan (korkealaatuisenakin) tehty ommel katsotaan esimerkiksi vaatekappaleessa virheeksi.

**Comment by jarmosaarikko** [comment]
katso myös aikasempi keskustelu kohdassa "viat" https://github.com/NatLibFi/Finto-data/issues/716
jossa käsiteltiin erilaisia virheitä

**Comment by YSO-kokous** [comment]
Pohdimme virheet-käsitteen hierarkisointia myöhemmin. Todettiin kuitenkin, että @mariaforsen laatii virheille uuden määritelmän ruotsiksi, jotta käsite erottuu viat-käsitteestä. Työstetään suomen kielen määritelmää tämän pohjalta.

**Comment by mirmaid** [comment]
Jos mietitään erottelua roolin ja objektin välillä siltä pohjalta mikä jokin on ensisijaisesti, en näe selkeää eroa kalusteella, näytteellä ja virheellä. Virhe voi olla ensisijaisesti väärässä paikassa sijaitseva ommel, näyte ensisijaisesti putkellinen verta ja kaluste ensisijaisesti tuoli. Jos taas tarkoitetaan sitä mitä varten jokin on ensisijaisesti tehty, ts. tuoli on tehty ensisijaisesti kalusteeksi, niin myös putkellinen verta on otettu ensisijaisesti näytteeksi. Ommel on toki ensisijassa tarkoitettu muuksi kuin virheeksi, mutta tarkoituksessa on epäonnistuttu ja ommel on tehty "liitoksen" sijasta virheeksi. 

Itselleni on aika vaikea myöskään hahmottaa objekti/rooli -eroa siinä, mikä on jonkin asian ensisijainen, tarkoitettu ja onnistuneesti toteutettu funktio (auto > kulkuneuvo) ja sen jokin sattumanvarainen muu funktio (auto > romu). Miksi toinen näistä olisi objekti ja toinen rooli? Tulee mieleeni taannoin katsomani muovi-dokumentin kommentti "Muovi tulisi nähdä alusta alkaen jätteenä." 

Tähän kaipaisin vielä selkeytystä niiltä, joiden mielestä rajanveto roolin ja objektin välillä ei ole vaikeaa.

**Comment by kseppala** [comment]
Pitää erottaa hierarkkiset ylä-alakäsitesuhteet kuten huonekalu/tuoli, kulkuneuvo/auto funktiosuhteista, jotka voivat ilmaista esim. käytön perusteella muodostuvan roolin. Auto on yhdenlainen kulkuneuvo, mutta kun se menee tarpeeksi huonoon kuntoon, siitä tulee romua. Veri on (yson mukaan) ensisijaisesti nestemäistä kudosta ja verinäyte on siis ensisijaisesti verta, se on vain otettu näytteeksi tutkimusta varten. Muovia ei tarkoituksella valmisteta jätteeksi, vaan ensisijaisesti muuhun tarkoitukseen käytettäväksi, mutta kun se poistetaan käytöstä, siitä tulee jätettä.

**Comment by mirmaid** [comment]
Ks. myös
#724
#586

**Comment by mirmaid** [comment]
Kartoitetaan käsitteitä, jotka voisivat olla rooleissa (tai rooli-käsitteitä, jotka voisivat olla muualla). Tarvitaan selkeä rajanveto siitä, mikä on rooli ja mikä ei.
